kmk/extensions/oled.py

```diff
@@ -0,0 +1,291 @@
+            width=self._width,
+            height=self._height,
+            rotation=180 if self._flip else 0,
+            brightness=self._brightness,
+        )
+        self.render_oled(0)
+        return
+
+    def before_matrix_scan(self, sandbox):
+        self.dim()
+        if self._dark != self._go_dark or sandbox.active_layers[0] != self._prevLayers:
+            self._dark = self._go_dark
+            self._prevLayers = sandbox.active_layers[0]
+            self.updateOLED(sandbox)
+        return
+
+    def after_matrix_scan(self, keyboard):
+        if keyboard.matrix_update or keyboard.secondary_matrix_update:
+            self.timer_time_reset()
+        return
+
+    def before_hid_send(self, sandbox):
+        return
+
+    def after_hid_send(self, sandbox):
+        return
+
+    def on_powersave_enable(self, sandbox):
+        self._powersave = True
+        return
+
+    def on_powersave_disable(self, sandbox):
+        self._powersave = False
+        return
+
+    def _oled_bri(self, *args, **kwargs):
+        self._display.brightness = (
+            self._display.brightness + self._brightness_step
+            if self._display.brightness + self._brightness_step <= 1.0
+            else 1.0
+        )
+        self._brightness = self._display.brightness  # Save current brightness
+
+    def _oled_brd(self, *args, **kwargs):
+        self._display.brightness = (
+            self._display.brightness - self._brightness_step
+            if self._display.brightness - self._brightness_step >= 0.1
+            else 0.1
+        )
+        self._brightness = self._display.brightness  # Save current brightness
+
+    def timer_time_reset(self):
+        self._timer_start = ticks_ms()
+
+    def dim(self):
+        if self._powersave:
+            if self._powersave_off_time_ms is not None and not check_deadline(
+                ticks_ms(), self._timer_start, self._powersave_off_time_ms
+            ):
+                self._go_dark = True
+            elif self._powersave_dim_time_ms is not None and not check_deadline(
+                ticks_ms(), self._timer_start, self._powersave_dim_time_ms
+            ):
+                self._display.brightness = (
+                    self._display.brightness / 2
+                    if self._display.brightness > 0.5
+                    else 0.1
+                )
+            else:
+                self._display.brightness = self._brightness
+                self._go_dark = False
+        elif self._off_time_ms is not None and not check_deadline(
+            ticks_ms(), self._timer_start, self._off_time_ms
+        ):
+            self._go_dark = True
+        elif self._dim_time_ms is not None and not check_deadline(
+            ticks_ms(), self._timer_start, self._dim_time_ms
+        ):
+            self._display.brightness = (
+                self._display.brightness / 2 if self._display.brightness > 0.5 else 0.1
+            )
```

Comment:
Again with the inline ifs. Makes it harder to read and doesn't even save loc, because the line gets too long.